x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichCache.java

```diff
@@ -206,29 +191,8 @@ private static Object innerDeepCopy(Object value, boolean unmodifiable) {
         }
     }
 
-    private static class CacheKey {
-
-        final String enrichIndex;
-        final SearchRequest searchRequest;
-
-        private CacheKey(String enrichIndex, SearchRequest searchRequest) {
-            this.enrichIndex = enrichIndex;
-            this.searchRequest = searchRequest;
-        }
-
-        @Override
-        public boolean equals(Object o) {
-            if (this == o) return true;
-            if (o == null || getClass() != o.getClass()) return false;
-            CacheKey cacheKey = (CacheKey) o;
-            return enrichIndex.equals(cacheKey.enrichIndex) && searchRequest.equals(cacheKey.searchRequest);
-        }
-
-        @Override
-        public int hashCode() {
-            return Objects.hash(enrichIndex, searchRequest);
-        }
-    }
+    // Visibility for testing
+    record CacheKey(String enrichIndex, int maxMatches, Object value) {}
 
     // Visibility for testing
     record CacheValue(List<Map<?, ?>> hits, Long sizeInBytes) {}
```

Comment:
I still didn't actually implement the improved cache size tracking, because the inaccuracy mostly comes from the cache key (which is already significantly reduced in size with this change, but not `0`) and I'd prefer to align first with others that this new cache key is in the direction we want to go in.

Reply:
I'm in favor of the new cache key -- I've wanted to fix this for a long time.